src/Hamiltonians/Sum.jl

```diff
@@ -0,0 +1,136 @@
+"""
+    HamiltonianSum(A::AbstractHamiltonian, B::AbstractHamiltonian; a::Number=1, b::Number=1, weight::Real=1)
```

Comment:
```suggestion
    HamiltonianSum(A::AbstractHamiltonian, B::AbstractHamiltonian; a=1, b=1, weight=1)
```
For readability I would leave out the type specification here. From context it is clear that these should be numbers. The code should later convert them to a sensible type.